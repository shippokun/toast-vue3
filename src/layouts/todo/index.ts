export { default as TodoLayout } from "./todo.layout.vue";
