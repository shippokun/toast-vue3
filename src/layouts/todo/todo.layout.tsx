import { defineComponent } from "vue";

export const TodoLayout = defineComponent({
  name: "TodoLayout",
  setup() {
    return () => (
      <div class="container is-max-desktop">
        <section class="hero">
          <div class="hero-body">
            <h1 class="title has-text-centered">Todo-Vue</h1>
          </div>
        </section>
        <router-view />
      </div>
    );
  },
});
