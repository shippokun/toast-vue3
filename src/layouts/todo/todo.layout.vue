<template>
  <div class="container is-max-desktop">
    <section class="hero">
      <div class="hero-body">
        <h1 class="title has-text-centered">Todo-Vue</h1>
      </div>
    </section>
    <router-view />
  </div>
</template>
<script lang="ts">
import { defineComponent } from "vue";

export default defineComponent({
  name: "TodoLayout",
});
</script>
