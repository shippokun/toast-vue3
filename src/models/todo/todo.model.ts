export interface Todo {
  id: number;
  title: string;
  context: string;
  complated: boolean;
  updatedAt: Date;
  createdAt: Date;
}

export interface TodoCreateDto {
  title: string;
  context: string;
  complated: boolean;
}
