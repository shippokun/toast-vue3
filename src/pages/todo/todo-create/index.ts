export { default as TodoCreatePage } from "./todo-create.page.vue";
