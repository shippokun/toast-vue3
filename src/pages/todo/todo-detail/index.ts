export { default as TodoDetailPage } from "./todo-detail.page.vue";
