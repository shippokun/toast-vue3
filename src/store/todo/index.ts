import { Module } from "vuex";

import { RootState } from "../store";
import { getters } from "./getters";
import { TodoState, initialState } from "./states";

export const module: Module<TodoState, RootState> = {
  namespaced: true,
  state: initialState,
  getters,
};

export * from "./getters";
export * from "./states";
