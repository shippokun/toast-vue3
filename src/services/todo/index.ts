import { TodoService } from "./todo.service";

export const todoService = new TodoService("http://localhost:3000");
