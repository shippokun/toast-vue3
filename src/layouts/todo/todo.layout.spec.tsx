import { shallowMount } from "@vue/test-utils";
import { TodoLayout } from "./todo.layout";

describe("todo.layout.tsx", () => {
  it("render", () => {
    const wrapper = shallowMount(TodoLayout);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
