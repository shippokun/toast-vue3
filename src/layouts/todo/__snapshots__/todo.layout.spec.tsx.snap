// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todo.layout.tsx render 1`] = `
<div class="container is-max-desktop">
  <section class="hero">
    <div class="hero-body">
      <h1 class="title has-text-centered">Todo-Vue</h1>
    </div>
  </section>
  <router-view></router-view>
</div>
`;
